### PR TITLE
Even more coverage! [WIP]

### PR DESCRIPTION
**Pull request by Lnaden**
This PR works to bring even more coverage of tests. I got utils coverage all the way up. However, I want to discuss the plan for PMF and some of the other choices I plan on making:

* The PMF module has many code paths which I can explore and test. Do we want to run down all of them? 
* For PMF methods without analytical errors, do we want to rely on the bootstrap to test them? I started doing this, but adding many bootstraps will significantly slow down the process.
* My IDE caught a bunch of syntax typos which were not hit in the tests, I'll try to fix them as I run into them.
* The Spline PMF test makes this call: 
    ```python
    pmf.generatePMF(u_kn, x_n,...)
    ```
    which uses `u_kn` instead of `u_n` like the histogram and KDE tests. I don't know if that was a typo so I wanted to ask.
* I assumed `numexpr` is required since we do require it in both the Conda env/recipe and in the setup.py
* I dropped the old `get_fn` which wasn't used and I think left over from copying the MDTraj utilities a long while back
* I plan on completely dropping `six` as a dependency, any objections?

### CODE REVIEW

**Comment by mrshirts** [comment]
> The PMF module has many code paths which I can explore and test. Do we want to run down all of them?

Some are more experimental, so I don't know that one needs to look at ALL of them.  I'll look over this now that you have updated the pytest stuff.

> For PMF methods without analytical errors, do we want to rely on the bootstrap to test them? I started doing this, but adding many bootstraps will significantly slow down the process.

I think that is very expensive.  My kludge is to use the histogram errors (which are fast), and should be reasonable for the amount of data collected. 

>which uses u_kn instead of u_n like the histogram and KDE tests. I don't know if that was a typo so I wanted to ask.

I don't think so, but I will check later when I look over the tests and explain why it's not  an error.

> I assumed numexpr is required since we do require it in both the Conda env/recipe and in the setup.py
It's used for logsumexp which is called by MBAR so I think so?

> I dropped the old get_fn which wasn't used and I think left over from copying the MDTraj utilities a long while back

Sounds good.

> I plan on completely dropping six as a dependency, any objections?

Sounds good.

**Comment by jaimergp** [comment]
> I plan on completely dropping six as a dependency, any objections?

Please 🙏

**Comment by Lnaden** [comment]
I think this is at the point where I've done what I can for the first pass. Theres lots of PMF codepaths which don't work for various reasons, and instead of having the tests fail for now, I just mark them `xfail` and we can circle back later. I've also cut back on the number of test permutations for speed sake, namely the `nbootstraps` which was adding a slow error computation to the KDE and Spline methods. 

This does Fix #351 in that I suppress the CodeCov warning for that collection from Patsy.

Six is no longer part of the repo now, yay!

**Comment by mrshirts** [comment]
> I think this is at the point where I've done what I can for the first pass.

Sounds good.  I will poke at it some more. 

> namely the nbootstraps which was adding a slow error computation to the KDE and Spline methods.

KDE is pretty fast, but spline is quite slow.  I'll look over it.  Again, the kluge is to use the histogram uncertainties, since most of the uncertainty is the data, not the method.  Since we we are generally just looking for LARGE deviations, getting the uncertainties off by 50-100% will not hurt. 

But why don't I evaluate this first, play around with it, and see what I think.

File: setup.cfg [hunk 1]
```diff
@@ -8,4 +8,9 @@ omit =
 description-file = README.md
 
 [aliases]
-test = pytest
\ No newline at end of file
+test = pytest
+
+[tool:pytest]
+filterwarnings =
+    ignore::DeprecationWarning:patsy.*
```
Comment (jaimergp) [inline setup.cfg]: Nice!

**Pull request merged**